app/helpers/application_helper/button/middleware_server_action.rb

```diff
@@ -1,7 +1,18 @@
 class ApplicationHelper::Button::MiddlewareServerAction < ApplicationHelper::Button::Basic
 
   def visible?
-    !@record.present? ||
-      (@record.try(:product) != 'Hawkular' && @record.try(:middleware_server).try(:product) != 'Hawkular')
+    @record.present? && !hawkular_product? && !immutable?
```

Comment:
Also, wouldn't EAP non-immutable servers have the Hawkular product if they are registered into miq by Hawkular? Also, what happens if the created provider for Hawkular is not called `Hawkular`, but something else? Does this field is unrelated to the created name?